website/integrations: add zendesk (#17541)
goauthentik/authentik: website/integrations/infrastructure/zendesk/index.mdx

> ---

Author: I have moved this to infrastructure. Thank you for the move to chat tho @dominic-r, was definitely better than misc

    - **Button text**: `authentik`

3. Click **Submit** to save.

  </TabItem>
  <TabItem value="saml">

## authentik configuration

To support the integration of Zendesk with authentik, you need to create an application/provider pair in authentik.

### Create an application and provider in authentik

1. Log in to authentik as an administrator and open the authentik Admin interface.
2. Navigate to **Applications** > **Applications** and click **Create with Provider** to create an application and provider pair. (Alternatively you can first create a provider separately, then create the application and connect it with the provider.)

- **Application**: provide a descriptive name, an optional group for the type of application, the policy engine mode, and optional UI settings. Take note of the **slug** as it will be required later.
- **Choose a Provider type**: select **SAML Provider** as the provider type.
- **Configure the Provider**: provide a name (or accept the auto-provided name), the authorization flow to use for this provider, and the following required configurations.
    - Note the **slug** value because it will be required later.

>     - Set the **ACS URL** to `https://zendesk.company/access/saml/`.

Reviewer: ```suggestion
    - Set the **ACS URL** to `https://company.zendesk.com/access/saml/`.
```

  ]}>
  <TabItem value="oidc">

## authentik configuration

To support the integration of Zendesk with authentik, you need to create an application/provider pair in authentik.

### Create an application and provider in authentik

1. Log in to authentik as an administrator and open the authentik Admin interface.
2. Navigate to **Applications** > **Applications** and click **Create with Provider** to create an application and provider pair. (Alternatively you can first create a provider separately, then create the application and connect it with the provider.)
    - **Application**: provide a descriptive name, an optional group for the type of application, the policy engine mode, and optional UI settings.
    - **Choose a Provider type**: select **OAuth2/OpenID Connect** as the provider type.
    - **Configure the Provider**: provide a name (or accept the auto-provided name), the authorization flow to use for this provider, and the following required configurations.
        - Note the **Client ID**, **Client Secret**, and **slug** values because they will be required later.
        - Set a `Strict` redirect URI to `https://company.zendesk.com/access/oidc/callback`.

>         - Select any available signing key.

Reviewer 2: ```suggestion
        - Under **Advanced Protocol Settings**, select an available signing certificate.
```

The other instances of this line are all OAuth integration docs where the signing **key** field is visible as soon as the provider settings page opens. 

On SAML providers the user has to expand the **Advanced Protocol Settings** section to see the signing **certificate** (not key) field.

Author: @dewi-tik this is true. But also, this guide is for both SAML and OIDC and this line is in the OIDC section

Author: lmao its all good these things get so confusing

>         - Select any available signing key.

Reviewer: Like in SAML we need to say where

Author: This is another copy paste, I searched across the repo and this wording is used 87 different files. So let's address that in a different PR

    - **Configure Bindings** _(optional)_: you can create a [binding](/docs/add-secure-apps/flows-stages/bindings/) (policy, group, or user) to manage the listing and access to applications on a user's **My applications** page.

3. Click **Submit** to save the new application and provider.

## Zendesk configuration

1. Log in to Zendesk as an administrator.
2. Navigate to **Security** > **Single sign-on (SSO)** > **Create SSO configuration** and click **OpenID Connect**.
    - **Configuration name**: `authentik`
    - **Client ID**: enter the client ID from authentik.
    - **Client secret**: enter the client secret from authentik.
    - **Scopes**: `openid profile email`
    - **Issuer**: `https://authentik.company/application/o/<application_slug>/`
    - **Authentication mode**: `PKCE`
    - **Issuer url**: `https://authentik.company/application/o/<application_slug>/`
    - **Show button when users sign in**: enabled
    - **Button text**: `authentik`

> 3. Click **Submit** to save.

Reviewer: ```suggestion
3. Click **Submit**.
```

Click submit to what? Anyways, this bit isn't really needed, as submitting you implies that you save the settings or something like that. You'd have to ask @tanberry for more clarification or whether it should be kept in this bit. Don't ask me why line 55 has the long form if the short one is preferred :)

### Create an application and provider in authentik

1. Log in to authentik as an administrator and open the authentik Admin interface.
2. Navigate to **Applications** > **Applications** and click **Create with Provider** to create an application and provider pair. (Alternatively you can first create a provider separately, then create the application and connect it with the provider.)

- **Application**: provide a descriptive name, an optional group for the type of application, the policy engine mode, and optional UI settings. Take note of the **slug** as it will be required later.
- **Choose a Provider type**: select **SAML Provider** as the provider type.
- **Configure the Provider**: provide a name (or accept the auto-provided name), the authorization flow to use for this provider, and the following required configurations.
    - Note the **slug** value because it will be required later.
    - Set the **ACS URL** to `https://company.zendesk.com/access/saml/`.
    - Set the **Issuer** to `https://authentik.company`.
    - Set the **Service Provider Binding** to `Post`.

>     - Under **Advanced protocol settings**, set **Signing Certificate** to use any available certificate.

Reviewer: I'm not sure I like this wording too much, I'll try to think of a better way to formulate it

3. Click **Submit** to save the new application and provider.

### Copy the Certificate Fingerprint (SHA256)

1. Log in to authentik as an administrator and open the authentik Admin interface.
2. Go to **System** > **Certificates** and click the **>** symbol next to the certificate pair you selected in the previous step.
3. Take note of the **Certificate Fingerprint (SHA256)** as it will be required in the next step.

## Zendesk configuration

1. Log in to Zendesk as an administrator.
2. Navigate to **Security** > **Single sign-on (SSO)** > **Create SSO configuration** and click **SAML**.
    - **Configuration name**: authentik
    - **SAML SSO URL**: `https://authentik.company/application/saml/<application_slug>/sso/binding/post/`
    - **Certificate fingerprint**: Paste the **Certificate Fingerprint (SHA256)** copied from authentik.
    - **Remote logout URL**: `https://authentik.company/application/saml/<application_slug>/slo/binding/redirect/`
    - **Show button when users sign in**: enabled
    - **Button value**: `authentik`

> 3. Click **Submit** to save.

Reviewer: same as 71

  </TabItem>
</Tabs>

## Configuration verification

> To confirm that authentik is properly configured with Zendesk, log out, then attempt to log back in by visit your Zendesk page and clicking **Login with authentik**.

Reviewer: ```suggestion
To confirm that authentik is properly configured with Zendesk, log out, then attempt to log back in by visit your Zendesk portal and clicking **Log in with authentik**.
```

- **Application**: provide a descriptive name, an optional group for the type of application, the policy engine mode, and optional UI settings. Take note of the **slug** as it will be required later.
- **Choose a Provider type**: select **SAML Provider** as the provider type.
- **Configure the Provider**: provide a name (or accept the auto-provided name), the authorization flow to use for this provider, and the following required configurations.
    - Note the **slug** value because it will be required later.
    - Set the **ACS URL** to `https://company.zendesk.com/access/saml/`.
    - Set the **Issuer** to `https://authentik.company`.
    - Set the **Service Provider Binding** to `Post`.
    - Under **Advanced protocol settings**, set **Signing Certificate** to use any available certificate.
- **Configure Bindings** _(optional)_: you can create a [binding](/docs/add-secure-apps/flows-stages/bindings/) (policy, group, or user) to manage the listing and access to applications on a user's **My applications** page.

3. Click **Submit** to save the new application and provider.

> ### Copy the signing certificate SHA256 fingerprint

Author: changed just let me know

1. Log in to authentik as an administrator and open the authentik Admin interface.

> 2. Go to **System** > **Certificates** > and click the certificate pair you selected in the previous step.

Reviewer: ```suggestion
2. Go to **System** > **Certificates** and click the certificate pair you selected in the previous step.
```

Author: This is true. I do not like this in our UI, but your statement is accruate. Updated
